### DISCUSSION TITLE
Error when requesting u_dot_old

### DISCUSSION THREAD

**Discussion by bylore** [idaholab/moose#25920] [Q&A General] [answered]
Hi, I was building my own material object which use **_dot_old** of a scalar field variable.
And I got the error

> MooseVariableFE: Old time derivative of solution (`u_dot_old`) is not stored. Please set uDotOldRequested() to true in FEProblemBase before requesting `u_dot_old`.

Even I already set uDotOldRequested by `_fe_problem.setUDotOldRequested(true); `

Here is my code in the construction function.
`  _fe_problem.setUDotRequested(true);
  _fe_problem.setUDotOldRequested(true); 
  _var_dot = &coupledDot("coupled_variable"); 
  _var_dot_old = &coupledDotOld("coupled_variable");`

Could you help me? Thanks

**Comment by GiudGiud** [comment, marked as answer]
Hello

This "setUDotRequested" is something that is done in the time integrator.
What time integrator are you using?

Guillaume

**Reply by bylore** [reply]
I use the default **implicit-euler** integrator of Transient Executioners.

**Reply by GiudGiud** [reply]
can you add the request for UDot in ImplicitEuler.C ? 
There's an example in NewmarkBeta.C for doing that

**Reply by bylore** [reply]
I found ImplicitEuler just inherits from TimeIntegrator where only UDotRequested is set True. Maybe I need build a new time intergrator inheriting from ImplicitEuler if I want ImplicitEuler and u_dot_old at the same time?

**Reply by bylore** [reply]
Yes, I will try it. Thank you~
